### ISSUE TITLE
Guidance on what database is best for Etherpad?

### ISSUE THREAD

**Issue by JamesMcMahon** [ether/etherpad-lite#4084]
The docs for the Docker image points to https://github.com/ether/ueberDB and says all databases that ueberDB supports are supported by Etherpad, which is great, but it puts me in a slightly awkward position as a operator where I have been given a lot of choice but no guidance. What are the tradeoffs involved for each database?

Are there docs / resources / information that could help me make an informed decision on which database to choose? If not, is that something that someone more familiar with Etherpad could put together?

**Comment by JohnMcLear** [comment]
There doesn't need to be guidance because **the database is not the limiting factor to performance.**

See https://travis-ci.org/github/ether/ueberDB

```
┌────────────────────┬──────────┬────────────────────┬───────────────┬───────────────┬───────────────┐

│ Database           │ # of it… │ Write(in seconds)  │ Read(scnds)   │ findKey(scnd… │ remove(scnds) │

├────────────────────┼──────────┼────────────────────┼───────────────┼───────────────┼───────────────┤

│ dirty:cache-on     │ 1000     │ 0.031              │ 0.001         │ 0.001         │ 0.001         │

├────────────────────┼──────────┼────────────────────┼───────────────┼───────────────┼───────────────┤

│ mysql:cache-on     │ 20000    │ 1.75585            │ 0.00315       │ 0.03          │ 0.0017        │

├────────────────────┼──────────┼────────────────────┼───────────────┼───────────────┼───────────────┤

│ postgres:cache-on  │ 20000    │ 1.78625            │ 0.003         │ 0.01925       │ 0.0013        │

├────────────────────┼──────────┼────────────────────┼───────────────┼───────────────┼───────────────┤

│ redis:cache-on     │ 20000    │ 1.76245            │ 0.00285       │ 0.01825       │ 0.00395       │

├────────────────────┼──────────┼────────────────────┼───────────────┼───────────────┼───────────────┤

│ dirty:cache-off    │ 1000     │ 0.032              │ 0.001         │ 0.001         │ 0.001         │

├──────────────���─────┼──────────┼────────────────────┼───────────────┼───────────────┼───────────────┤

│ mysql:cache-off    │ 20000    │ 1.85145            │ 0.0032        │ 0.04375       │ 0.00085       │

├────────────────────┼──────────┼────────────────────┼───────────────┼───────────────┼───────────────┤

│ postgres:cache-off │ 20000    │ 1.83825            │ 0.0026        │ 0.0081        │ 0.00205       │

├────────────────────┼──────────┼────────────────────┼───────────────┼───────────────┼───────────────┤

│ redis:cache-off    │ 20000    │ 1.87335            │ 0.0027        │ 0.0154        │ 0.0012        │

└────────────────────┴──────────┴────────────────────┴──────���────────┴───────────────┴───────────────┘
```

TLDR; 

1. Dirty is not for production.
1. Other database are very similar in terms of performance.

If you want further testing / performance / metrics see: https://github.com/ether/ueberDB/blob/dirtyref/test/test.js

Some databases lack features / have limitations, see https://github.com/ether/ueberDB/tree/dirtyref#feature-support

**Comment by JohnMcLear** [comment]
Just to add to this, I "think" the limiting factor is socketIO, at some point I want to try something alternative to socketio to prove my theory.

**Comment by JamesMcMahon** [comment]
@JohnMcLear Thanks for the reply! Even saying that all databases are roughly the same performance wise is pretty good guidance.

To give a little more context here, I have a working [Helm chart](https://github.com/focused-labs/etherpad-helm) for Etherpad that I am looking to clean up and circulate more widely. When I was working on it I hit a few surprises with the DB, well surprises may be the wrong word, but I hit some Mysql Charset issues, which seem to be pretty well documented, and I attempted to use Mongo to find it's not included on the official Docker image. We ended up using Postgres because that worked will with the Docker image but I wanted to get some feedback / advice from the Etherpad community to validate the Postgres decision.

**Comment by JohnMcLear** [comment]
MySQL / María should be fine.  Sounds like the docker community will need to work on the MySQL Etherpad docker image?

Postgres should be fine too.

Generally my rule is to work on the software not any containerization and leave the countable communities to do their thing.

Happy to help if needed tho!
